addons/Drawing/src/presenter.js

```diff
@@ -1420,5 +1477,45 @@ function AddonDrawing_create() {
         presenter.closeTextFieldPopup();
     };
 
+    function sendModifiedEvent() {
+        if (blockNextSendOfModifiedEvent) {
+            blockNextSendOfModifiedEvent = false;
+            return;
+        }
+
+        if (!presenter.isModified) {
+            return;
+        }
+        presenter.sendModifiedEvent();
+    }
+
+    presenter.sendModifiedEvent = function () {
+        const eventData = {
+            'source': presenter.configuration.addonID,
+            'item': '',
+            'value': 'modified',
+            'score': ''
+        };
+
+        eventBus.sendEvent('ValueChanged', eventData);
+    };
+
+    presenter.sendEmptyEvent = function () {
+        const eventData = {
+            'source': presenter.configuration.addonID,
+            'item': '',
+            'value': 'empty',
+            'score': ''
+        };
+
+        eventBus.sendEvent('ValueChanged', eventData);
+    };
+
+    presenter.preDestroy = function() {
+        if (!presenter.shouldUpdateState) {
```

Comment:
Proszę o poprawę na 
`if (!presenter.isModified) {`
Rozumiem dlaczego skorzystałeś ze zmiennej presenter.shouldUpdateState jednak event powinien być wysyłany gdy nie było modyfikacji w canvas za co odpowiada presenter.isModified. presenter.shouldUpdateState odpowiada za to, czy stan powinien zostać zaktualizowany czyli gdy np. zmieniono kolor, ustawiono gumkę itd.